.azure-pipelines/recover_testbed/common.py

```diff
@@ -66,20 +66,25 @@ def posix_shell_onie(dut_console, mgmt_ip, image_url):
 
                     x = x.decode('ISO-8859-9')
 
-                    if "GNU GRUB" in x:
-                        enter_onie_flag += 1
+                    if is_nexus and "loader" in x and ">" in x:
+                        dut_console.remote_conn.send('reboot\n')
                         continue
 
-                    if "SONiC-OS-" in x and enter_onie_flag == 1:
+                    # if "GNU GRUB" in x:
+                    #     enter_onie_flag += 1
+                    #     continue
+
+                    if "-OS-" in x and enter_onie_flag is True:
```

Comment:
if "-OS-" in x and enter_onie_flag:

Reply:
I do some tests, and because of the rule of Python, we can use `bool(enter_onie_flag)` or use `enter_onie_flag is True` here. Simply `enter_onie_flag` will not consider as a boolean.